src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md

```diff
@@ -0,0 +1,330 @@
+| + , - , \|                    | additive operation               |
+| == , != , < , > , <= , >=     | comparison operation             |
+| and                           | logical conjunction              |
+| or                            | logical disjunction              |
+
+## Arithmetic expressions
+
+Arithmetic expressions are defined for real and integer data types only. These
+data types may be mixed for any given expression.
+
+| Operator | Description                     |
+|----------|---------------------------------|
+| *        | multiplication                  |
+| /        | division                        |
+| %        | remainder                       |
+| +        | addition                        |
+| -        | subtraction/additive inverse    |
+
+### Notes
+
+* If any sub expression in an arithmetic expression is real, the expression will
+evaluate to a data type of real.  
+* Division of two integers results in the integer division result (truncated)
+  and will not produce a real value.  
+* When `-` is used as a unary operation, it evaluates to the additive inverse of
+  the operand.  
+
+### Examples
+```
+  -27
+  2 + 2
+  (x + y) / 2
+  0.707 * voltage
+  (plane.offset + ALT::get_altitude())
+```
+
+## Boolean expressions
+
+A boolean expression is any expression that evaluates to either a `true` or
+`false` value. Boolean expressions are often used for comparison in statements
+like `if` and `while`, and also in `where` clauses. Although boolean expressions
+usually contain other expression types (such as arithmetic or string
+expressions), they can also be used to compare time values, instance references,
+instance set references, and unique IDs. There is also one unary operator,
+`not`, which can be used to logically negate a boolean expression.
+
+| Operator    | Description                      | Valid operand types                        |
+|-------------|----------------------------------|--------------------------------------------|
+| ==          | equality                         | any                                        |
+| !=          | inequality                       | any                                        |
+| <           | less than                        | integer, real, date, timestamp, string     |
+| >           | greater than                     | integer, real, date, timestamp, string     |
+| <=          | less than or equal to            | integer, real, date, timestamp, string     |
+| >=          | greater than or equal to         | integer, real, date, timestamp, string     |
+| and         | logical conjunction              | boolean                                    |
+| or          | logical disjunction              | boolean                                    |
+| not         | boolean negation                 | boolean                                    |
+| \*empty     | instance reference/set empty     | instance reference, instance reference set |
```

Comment:
I think this and the next should be replaced with empty<sup>1</sup> and then number the reference at the end of the table.  The asterisk he could be confused as part of the name of the operator.

Reply:
done